### PR TITLE
Enable PIN prompt for netcore

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes: https://github.com/NuGet/Home/issues/9555
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details: 
Enable the PIN prompt for .NET Core, by setting silent flag to false.
For .NET Framework, the silent parameter is not used and a PIN prompt is always shown.

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  
Validation:

### CODE REVIEW

**Comment by nkolev92** [comment]
linked issue? 
Is this the port of https://github.com/NuGet/NuGet.Client/pull/3069?

**Review by zivkan — DISMISSED** [review]

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs [hunk 1]
```diff
@@ -136,7 +136,12 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
+#if IS_DESKTOP
                 cms.ComputeSignature(cmsSigner);
+#else
+            // In .NET Framework, this parameter is not used and a PIN prompt is always shown. In .NET Core, the silent flag needs to be set to false to show a PIN prompt.
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs]: nit: indentation level

**Comment by heng-liu** [comment]
> linked issue?
> Is this the port of #3069?

Yes. Added into the description. Thanks Nikolche!

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs (hunk 1 above)
Comment (heng-liu) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs]: Thanks! Fixed.

**Comment by nkolev92** [comment]
@heng-liu that's a PR link :D 
I was asking for an issue :D 

fyi @clairernovotny

**Comment by heng-liu** [comment]
@nkolev92  oops, yes it's a PR.. Shall I create an issue for that and add it to the link?

**Comment by nkolev92** [comment]
Yes, remember https://github.com/NuGet/NuGet.Client/blob/dev/CONTRIBUTING.md#contributing-step-by-step and https://github.com/NuGet/NuGet.Client/blob/dev/.github/PULL_REQUEST_TEMPLATE.md. 

The pull request template specifically has the following: 

```
Fixes: Link_to_issue (search NuGet/Home, and create one if you can't find a suitable issue) 
```

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs [hunk 2]
```diff
@@ -136,7 +136,12 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
+#if IS_DESKTOP
                 cms.ComputeSignature(cmsSigner);
+#else
+                // In .NET Framework, this parameter is not used and a PIN prompt is always shown. In .NET Core, the silent flag needs to be set to false to show a PIN prompt.
+                cms.ComputeSignature(cmsSigner, false);
```
Comment (dominoFire) [inline src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs]: Suggestion: add a parameter name to the second argument

Comment (heng-liu) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs]: Thanks for reviewing! Fixed.

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs [hunk 3]
```diff
@@ -136,7 +136,12 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
+#if IS_DESKTOP
                 cms.ComputeSignature(cmsSigner);
+#else
+                // In .NET Framework, this parameter is not used and a PIN prompt is always shown. In .NET Core, the silent flag needs to be set to false to show a PIN prompt.
+                cms.ComputeSignature(cmsSigner, silent: false);
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:143]: As per [doc](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.pkcs.signedcms.computesignature?view=net-5.0), the difference I see between these overloaded methods is `cms.ComputeSignature(cmsSigner, silent: false);` call can throw `PlatformNotSupportedException`. Do we need to catch this exception along with `CryptographicException`?

**Comment by zivkan** [comment]
I was going to take a look, but since the destination branch was changed, this PR is now showing 165 files changes. Can you please rebase it?

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs (hunk 3 above)
Comment (zivkan) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:143]: The doc says that `PlatformNotSupportedException` is thrown when "A signing certificate is not specified" on .NET Core, but on .NET Framework the same condition throws `InvalidOperationException`. We're not catching `InvalidOperationException` here, so the behaviour is consistent between .NET Core and .NET Framework.

**Review by zivkan — APPROVED** [review]

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs [hunk 4]
```diff
@@ -136,7 +136,12 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
...
+#endif
             }
             catch (CryptographicException ex) when (ex.HResult == INVALID_PROVIDER_TYPE_HRESULT)
```
Comment (kartheekp-ms) [inline src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:146]: I think this is out of scope for this PR. may be we need a tracking issue to avoid depending on `HRResult`.

**Review by kartheekp-ms — APPROVED** [review]

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs (hunk 4 above)
Comment (heng-liu) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:146]: Thanks! https://github.com/NuGet/Home/issues/9628 raised.
